needy/library.py

```diff
@@ -176,16 +173,21 @@ def __write_build_status(self):
             json.dump(status, status_file)
 
     def __cache_artifacts(self):
-        if self.__build_cache:
+        if self.__build_caches:
             try:
-                self.__build_cache.store_artifacts(self.build_directory(), self.__cache_key())
+                self.__build_caches[0].store_artifacts(self.build_directory(), self.__cache_key())
```

Comment:
For now. It's a semantic argument and I think the correct behavior is to cycle through to the first writable cache and try it. Currently, all caches are writable, so I'm thinking there needs to be a configuration option, `disable-write`, or similar. This way, maybe you have a shared developer s3 cache with writing disabled for everyone but the ci server.